sql/dd/dd_table.cc

```diff
@@ -2332,10 +2332,17 @@ static std::unique_ptr<dd::Table> create_dd_system_table(
   }
 
   // Register the se private id with the DDSE.
-  handlerton *ddse = get_dd_engine(thd);
-  if (ddse->dict_register_dd_table_id == nullptr) return nullptr;
-  ddse->dict_register_dd_table_id(tab_obj->se_private_id());
-
+  if (opt_initialize) {
+    handlerton *ddse = get_dd_engine(thd);
+    if (ddse->dict_register_dd_table_id == nullptr) return nullptr;
+    ddse->dict_register_dd_table_id(tab_obj->se_private_id());
+  } else {
+    assert(file->ht->db_type == DB_TYPE_INNODB ||
+           file->ht->db_type == DB_TYPE_ROCKSDB);
+    if (file->ht == nullptr || file->ht->dict_register_dd_table_id == nullptr)
```

Comment:
`file->ht == nullptr` check comes after unconditional dereference in the assert above, either the assert needs fixing, either (more likely IMHO) the first condition has to be dropped